libclc/amdgcn-amdhsa/libspirv/math/lgamma.cl

```diff
@@ -8,12 +8,23 @@
 
 #include <clcmacro.h>
 #include <spirv/spirv.h>
- 
-double __ocml_lgamma_f64(double);
-float __ocml_lgamma_f32(float);
- 
+
 #define __CLC_FUNCTION __spirv_ocl_lgamma
 #define __CLC_BUILTIN __ocml_lgamma
+
+float __ocml_lgamma_f32(float);
 #define __CLC_BUILTIN_F __CLC_XCONCAT(__CLC_BUILTIN, _f32)
+
+#ifdef cl_khr_fp64
+#pragma OPENCL EXTENSION cl_khr_fp64 : enable
```

Comment:
probably a nit but I don't understand this: why enable an OPENCL EXTENSION on a non opencl backend? Presumably the `cl_khr_fp64` was already working before this patch so I don't imagine this changes any behaviour.

Reply:
This has been resolved in internal discussions already, but I'd still like to post it here as it might be interesting for others as well.
The fact that the AMDGPU target is a non-opencl doesn't matter here.
We basically enable the OpenCL  fp64 extension because libclc is compiled by an OpenCL C compiler (i.e. it needs to be compliant with the openCL language specification).
Not enabling it would lead to the fact that a fp64 literal declared inside this cl file could be compiled as a float.
This is basically where Vanilla OpenCL C deviates from the C standard.
Potentially, this could lead to unexpected behaviour in the future and therefore, it's safer to enable the extension.

For the fp16 extension that is also enabled several times in cl files by this patch, it's a different story.
Not enabling it (i.e. removing the pragmas) leads to errors at compile time with the error message that the `half` type is not allowed.
It's needed to actually make it a type:
https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#reinterpreting-types-using-as_type-and-as_typen
(footnote 15)

This is also why @jinz2014 added these pragmas and wrapped them inside the ifdef macro checks